crates/runtime/src/dataconnector/github/mod.rs

```diff
@@ -158,9 +160,32 @@ impl Github {
             provider_builder
         };
 
+        // Add a sanity check to ensure the endpoint exists
+        let inspection_query_string = format!(
+            r#"{{
+            repositoryCheck: repository(owner: "{owner}", name: "{repo}") {{
+                id
+                nameWithOwner
+            }}
+        }}"#,
+        );
+
+        let query_arc = Arc::from(inspection_query_string);
+        let mut inspection_query = GraphQLQuery::try_from(query_arc).map_err(|e| {
```

Comment:
There isn't a way to specify the json pointer directly on here? i.e. GraphQLQuery::try_from(query_arc).map_err(..).with_json_pointer()?

Reply:
If not we should add it.